src/js/Calculator.js

```diff
@@ -0,0 +1,164 @@
+import {
+  OPERATOR_ADD,
+  OPERATOR_DIV,
+  OPERATOR_MUL,
+  OPERATOR_SUB,
+  REGEXP_OPERATOR,
+  INVALID_LENGTH,
+  REQUIRED_DIGIT,
+  OPERATORS_SET,
+  REGEXP_DIGIT,
+  OPERATOR_ZERO,
+  OPERATOR_AC,
+  OPERATOR_EQU
+} from "./constants.js";
+
+class Validator {
+  isNotKeypad(target) {
+    return !['.digit', '.modifier', '.operation'].some(key => target.matches(key));
+  }
+
+  isDigit(keyword) {
+    return REGEXP_DIGIT.test(keyword);
+  }
+
+  isOperator(keyword) {
+    return OPERATORS_SET.has(keyword);
+  }
+
+  isOnlyZero(keyword) {
+    return keyword === OPERATOR_ZERO;
+  }
+
+  isAC(keyword) {
+    return keyword === OPERATOR_AC;
+  }
+
+  isEquation(keyword) {
+    return keyword === OPERATOR_EQU;
+  }
+
+  isFirstDigitInput(keyword, display) {
+    return this.isOnlyZero(display) && this.isDigit(keyword);
+  }
+
+  isDuplicatedOperator(keyword, display) {
+    if (this.isOnlyZero(display)) return true;
+    const lastInput = display.charAt(display.length - 1);
+    return [keyword, lastInput].every(key => this.isOperator(key));
+  }
+
+  isOverOperandLength(keyword, expressions) {
+    if (this.isOperator(keyword)) return false;
+    const firstOperand = expressions[0];
+    const lastOperand = expressions.at(-1);
+    if (lastOperand === undefined) return `${firstOperand}${keyword}`.length > 3;
+    return `${lastOperand}${keyword}`.length > 3;
+  }
+
+  isAddition(operator) {
+    return operator === OPERATOR_ADD;
+  }
+
+  isSubtraction(operator) {
+    return operator === OPERATOR_SUB;
+  }
+
+  isValidOperand(operator, operand) {
+    if (this.isAddition(operator) || this.isSubtraction(operator)) return operand;
+    return operand || 1;
+  }
+}
+
+class Calculator {
+  #validator = new Validator();
+  #statement = [];
+  #priorityOperator = {
+    [OPERATOR_ADD]: 1,
+    [OPERATOR_SUB]: 1,
+    [OPERATOR_MUL]: 2,
+    [OPERATOR_DIV]: 2
+  }
+  #operatingSystem = new Map([
+    [OPERATOR_ADD, (left, right) => left + right],
+    [OPERATOR_SUB, (left, right) => left - right],
+    [OPERATOR_MUL, (left, right) => left * right],
+    [OPERATOR_DIV, (left, right) => Math.truc(left / right)],
```

Comment:
```suggestion
    [OPERATOR_DIV, (left, right) => Math.trunc(left / right)],
```
테스트를 돌려봤는데 나눗셈이 안되는 현상이 있어요.

원래 의도하신게 trunc 같아요.
trunc이 정수만 반환하는 함수네요. 처음보는 메서드라 신기했어요 배워갑니다 👍